tests/api/lists/add_selections.test.js

```diff
@@ -0,0 +1,78 @@
+const { getUserB, shouldNotBeCalled, rethrowShouldNotBeCalledErrors } = require('tests/api/utils/utils')
+const { authReq } = require('../utils/utils')
+const { createList } = require('../fixtures/lists')
+const { createEdition, someFakeUri } = require('../fixtures/entities')
+
+const endpoint = '/api/lists?action='
+const addSelection = `${endpoint}add-selections`
+const byIds = `${endpoint}by-ids&with-selections=true`
+
+describe('lists:add-selections', () => {
+  it('should reject without list id', async () => {
+    try {
+      await authReq('post', addSelection).then(shouldNotBeCalled)
+    } catch (err) {
+      rethrowShouldNotBeCalledErrors(err)
+      err.body.status_verbose.should.equal('missing parameter in body: id')
+      err.statusCode.should.equal(400)
+    }
+  })
+
+  it('should reject without uri', async () => {
+    const { list } = await createList()
+    try {
+      await authReq('post', addSelection, {
+        id: list._id
+      })
+      .then(shouldNotBeCalled)
+    } catch (err) {
+      rethrowShouldNotBeCalledErrors(err)
+      err.body.status_verbose.should.equal('missing parameter in body: uris')
+      err.statusCode.should.equal(400)
+    }
+  })
+
+  it('should reject creating a selection with an unknown entity', async () => {
+    const { list } = await createList()
+    await authReq('post', addSelection, {
+      id: list._id,
+      uris: [ someFakeUri ]
+    })
+    .then(shouldNotBeCalled)
+    .catch(err => {
+      err.statusCode.should.equal(403)
+      err.body.status_verbose.should.equal('entities not found')
+    })
+    const res = await authReq('get', `${byIds}&ids=${list._id}`)
+    const firstList = res.lists[list._id]
+    firstList.selections.length.should.equal(0)
+  })
+
+  it('should add uri and create selection', async () => {
+    const { list } = await createList()
+    const { uri } = await createEdition()
+    await authReq('post', addSelection, {
+      id: list._id,
+      uris: [ uri ]
+    })
+    const res = await authReq('get', `${byIds}&ids=${list._id}`)
+    const firstList = res.lists[list._id]
+    firstList.selections[0].uri.should.equal(uri)
+  })
+
```

Comment:
a test do set the behavior of trying to add a list element with a uri that is already in the list would be welcome

Reply:
done in 43c0c887